arelle/plugin/validate/ESEF/__init__.py

```diff
@@ -82,63 +97,70 @@
 docTypeXhtmlPattern = re.compile(r"^<!(?:DOCTYPE\s+)\s*html(?:PUBLIC\s+)?(?:.*-//W3C//DTD\s+(X?HTML)\s)?.*>$", re.IGNORECASE)
 
 FOOTNOTE_LINK_CHILDREN = {qnLinkLoc, qnLinkFootnoteArc, qnLinkFootnote, qnIXbrl11Footnote}
-PERCENT_TYPE = qname("{http://www.xbrl.org/dtr/type/numeric}num:percentItemType")
+PERCENT_TYPE = qname("{http://www.xbrl.org/dtr/type/numeric}num:percentItemType") #  type: ignore[no-untyped-call]
 IXT_NAMESPACES = {ixtNamespaces["ixt v4"], # only tr4 or newer REC is currently recommended
                   ixtNamespaces["ixt v5"]}
 
-def etreeIterWithDepth(node, depth=0):
+def etreeIterWithDepth(node: ModelObject | _Element, depth: int=0): # type: ignore[no-untyped-def]
```

Comment:
My typing skills are not yet strong enough to handle functions that yield. I hope to become better so that I can fix this.

Reply:
Looks good, thanks!

Reply:
Oh wait, it took a while for VS code to process after I changed it. Getting 94 new errors now.
```
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "None" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "str" of "Optional[str]" has no attribute "contextID"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "None" of "Optional[str]" has no attribute "contextID"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "str" of "Optional[str]" has no attribute "value"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:825: error: Item "None" of "Optional[str]" has no attribute "value"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:830: error: Item "str" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:830: error: Item "None" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:864: error: Item "str" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:864: error: Item "None" of "Optional[str]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:881: error: Item "_Element" of "Union[ModelObject, _Element]" has no attribute "qname"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:881: error: "_Element" has no attribute "qname"  [attr-defined]
arelle/plugin/validate/ESEF/__init__.py:919: error: Assignment to variable "err" outside except: block  [misc]
arelle/plugin/validate/ESEF/__init__.py:1049: error: Item "str" of "Optional[str]" has no attribute "concept"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:1049: error: Item "None" of "Optional[str]" has no attribute "concept"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:1053: error: Item "str" of "Optional[str]" has no attribute "unit"  [union-attr]
arelle/plugin/validate/ESEF/__init__.py:1053: error: Item "None" of "Optional[str]" has no attribute "unit"  [union-attr]
```

Reply:
I'm not sure I want to go down this rabbit hole for the moment. It's probably a lot easier to fix when we've added more type hints throughout the code.